### PR TITLE
*: add check index exist.

### PR DESCRIPTION
**Pull request by qiuyesuifeng**

### CODE REVIEW

**Comment by siddontang** [comment]
any test?

File: kv/index_iter.go [hunk 1]
```diff
@@ -246,3 +244,30 @@ func (c *kvIndex) SeekFirst(txn Transaction) (iter IndexIterator, err error) {
...
+	}
+
+	// For uniq index, the value of key is handle.
+	if c.unique {
+		handle, err := decodeHandle(value)
+		if err != nil {
+			return false, errors.Trace(err)
+		}
+
+		return h == handle, nil
```
Comment (siddontang) [inline kv/index_iter.go]: If duplicated, I think return the duplicated handler may be better for outer use. 

so maybe the definition may be Get() (handler int64, err error)

Comment (shenli) [inline kv/index_iter.go]: +1
For OnDuplicateKey insert, we need to get the duplicate row's key and update that row.

**Comment by zimulala** [comment]
LGTM

**Comment by siddontang** [comment]
LGTM

**Pull request merged**